include/solvers/mpm_base.tcc

```diff
@@ -762,16 +762,44 @@ bool mpm::MPMBase<Tdim>::initialise_math_functions(const Json& math_functions) {
       const std::string function_type =
           function_props["type"].template get<std::string>();
 
+      // Initiate another function_prop to be passed
+      auto function_props_update = function_props;
+
+      // Create a file reader
+      const std::string io_type =
+          io_->json_object("mesh")["io_type"].template get<std::string>();
+      auto reader = Factory<mpm::IOMesh<Tdim>>::instance()->create(io_type);
+
+      // Math function is specified in a file, replace function_props_update
+      if (function_props.find("file") != function_props.end()) {
+        // Make separate arrays
+        std::vector<double> xvalues;
```

Comment:
Having separate vectors makes it decoupled between x and y values, use a vector of arrays instead.

Reply:
OK